pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -126,18 +98,72 @@ def _update_next_infection_status(self, person: Person):
             Instance of person class with infection status attributes
 
         """
+        if person.infection_status in [InfectionStatus.Recovered,
+                                       InfectionStatus.Dead]:
+            person.next_infection_status = None
+        else:
+            row_index = person.infection_status.name
+            weights = self.state_transition_matrix.loc[row_index].to_numpy()
+            outcomes = range(1, self.number_of_states + 1)
+
+            if len(weights) != len(outcomes):
+                raise AssertionError('The number of infection statuses must \
+                                    match the number of transition \
+                                    probabilities')
```

Comment:
Is this aligned right? And we generally avoid backslash as a line break where possible - string addition is preferred here